app/models/user_group.rb

```diff
@@ -82,9 +83,25 @@ def identity?
     !!memberships.where(identity: true).pluck(:identity).first
   end
 
+  def verify_join_token(token_to_verify)
+    join_token.present? && join_token == token_to_verify
+  end
+
+  def public?
```

Comment:
This and `#member?` are no longer needed after having removed the old behaviour. Remove them.